toolkit/docs/building/building.md

```diff
@@ -71,58 +77,119 @@ Quick Start
+### 1) Bootstrap Toolchain
+A set of bootstrapped toolchain packages (gcc etc.) are used to build CBL-Mariner packages and images.
+
+```bash
+# Download the pre-built toolchain packages (REBUILD_TOOLCHAIN=n is the default value)
+sudo make toolchain REBUILD_TOOLS=y
+# Move the downloaded bootstrap packages to the general RPM out folder
+sudo make copy-toolchain-rpms
+```
+NOTE: A full list of targets and options to `make` is available [here](#all-build-targets) and [here](#all-build-variables).
+
+#### Full toolchain Bootstrap From Sources
+If you want to build **everything** from scratch, including the bootstrapping process, run:
 ```bash
 cd ~/git/CBL-Mariner/toolkit
+# Do a FULL bootstrap + rebuild from sources instead (much slower)
+# Add REBUILD_TOOLCHAIN=y to any subsequent command to ensure locally built toolchain packages are used
 sudo make toolchain REBUILD_TOOLS=y REBUILD_TOOLCHAIN=y DOWNLOAD_SRPMS=y
 ```
+This will download the source files (SRPMs) from the package sever, and build them locally.
+See the detailed section on building from scratch [here](#building-everything-from-scratch)
 
-NOTE: A full list of targets and options to sudo make is available [here](#all-build-targets) and [here](#all-build-variables).
 
-## Build Packages
-Once the toolchain is built, all packages can be built with the toolchain.  Large parts of the build are parallelized. Enable this by setting the `-j` flag for `make` to the number of parallel jobs to allow. (Recommend setting this value to the number of hyper-threads available on your system, or less)
+### 2) Build All Packages
+(**this step may be omitted if desired**)
 
-```bash
-# Still from the toolkit folder, copy the toolchain archive built in the previous step into the toolkit folder
-mv ../build/toolchain/toolchain_built_rpms_all.tar.gz .
+Once the toolchain is bootstrapped, packages can be built with the toolchain.
+
+The image build commands in [3) Build Images](#3-build-images) will **automatically** build only the required packages based on the selected configuration.
 
+However, to manually build **all** packages you can clear the configuration with `CONFIG_FILE=` and invoke the package build target.
+
+Large parts of the build are parallelized. Enable this by setting the `-j` flag for `make` to the number of parallel jobs to allow. (Recommend setting this value to the number of logical cores available on your system, or less)
+
+**NOTE: If you are building your toolchain packages from source, add `REBUILD_TOOLCHAIN=y`**
+```bash
 # Build ALL packages FOR AMD64
-sudo make build-packages -j$(nproc) CONFIG_FILE= TOOLCHAIN_ARCHIVE=toolchain_built_rpms_all.tar.gz DOWNLOAD_SRPMS=y REBUILD_TOOLS=y REBUILD_TOOLCHAIN=n REBUILD_PACKAGES=y PACKAGE_IGNORE_LIST="openjdk8 openjdk8_aarch64 shim-unsigned-aarch64"
+sudo make build-packages -j$(nproc) CONFIG_FILE= DOWNLOAD_SRPMS=y REBUILD_TOOLS=y
```

Comment:
This needs to be restored unless the build tools were fixed.  I do not think this was fixed yet.  Maybe it's ok on 1.0-dev....but it was absolutely broken on 1.0.